### PR TITLE
feat: publish docker with a tagged release

### PR DESCRIPTION
**Pull request by douglaz**
Untested, I hope it works

### CODE REVIEW

File: .github/workflows/ci-nix.yml [hunk 1]
```diff
@@ -305,6 +305,14 @@ jobs:
           nix_tag=${{ env.gatewayd_container_tag }} && hub_tag="fedimint/gatewayd:master" && docker tag "$nix_tag" "$hub_tag" && docker push "$hub_tag"
           nix_tag=${{ env.gateway-cli_container_tag }} && hub_tag="fedimint/gateway-cli:master" && docker tag "$nix_tag" "$hub_tag" && docker push "$hub_tag"
 
+      - name: Publish tagged release
+        if: github.ref_type == 'tag'
```
Comment (wbobeirne) [inline .github/workflows/ci-nix.yml:309]: If I understand correctly, this _shouldn't_ work because the workflow runs on branch push, not tag push.

Comment (douglaz) [reply .github/workflows/ci-nix.yml:312]: Added to push:
```yaml
    tags:
      - 'v*'
```

**Review by dpc — APPROVED** [review]

**Comment by dpc** [comment]
Nothing obviously wrong, let's try. :D

File: .github/workflows/ci-nix.yml [hunk 2]
```diff
@@ -305,6 +307,14 @@ jobs:
...
+        run: |
+          nix_tag=${{ env.fedimintd_container_tag }} && hub_tag="fedimint/fedimintd:${GITHUB_REF_NAME}" && docker tag "$nix_tag" "$hub_tag" && docker push "$hub_tag"
+          nix_tag=${{ env.fedimint_cli_container_tag }} && hub_tag="fedimint/fedimint-cli:${GITHUB_REF_NAME}" && docker tag "$nix_tag" "$hub_tag" && docker push "$hub_tag"
+          nix_tag=${{ env.gatewayd_container_tag }} && hub_tag="fedimint/gatewayd:${GITHUB_REF_NAME}" && docker tag "$nix_tag" "$hub_tag" && docker push "$hub_tag"
+          nix_tag=${{ env.gateway-cli_container_tag }} && hub_tag="fedimint/gateway-cli:${GITHUB_REF_NAME}" && docker tag "$nix_tag" "$hub_tag" && docker push "$hub_tag"
```
Comment (wbobeirne) [inline .github/workflows/ci-nix.yml:316]: Looks like this failed (https://github.com/fedimint/fedimint/actions/runs/6016426153/job/16320403586):
```
denied: requested access to the resource is denied
Error: Process completed with exit code 1.
```
Looking at the run command, I don't see `GITHUB_REF_NAME` in the env list:
```
env:
    LAST_COMMIT_SHA: d[8](https://github.com/fedimint/fedimint/actions/runs/6016426153/job/16320403586#step:14:8)d358[9](https://github.com/fedimint/fedimint/actions/runs/6016426153/job/16320403586#step:14:9)1982bdd34d5e320d1e6cef8ca877e32f8
    NIX_PATH: nixpkgs=channel:nixos-22.05
    fedimintd_container_tag: fedimintd:ic7ldl1437zgc49kn8nbr29k1nz35r9y
    fedimint_cli_container_tag: fedimint-cli:nqfai3yr480ig8ivvnkfj68zwqyr2qyq
    gatewayd_container_tag: gatewayd:v5i6sy2yq31jwx72ibag2sqj2m8d1h1w
    gateway-cli_container_tag: gateway-cli:dzskmz7pa4b651abzvyc3w4sz23fhqkr
```
Should we use `${{  github.ref_name }}` instead?

**Pull request merged**